### ISSUE TITLE
Days Until/Since

### ISSUE THREAD

**Issue by eatondpe** [roamhacker/SmartBlocks#69]
## ✂️ Copy of your #42SmartBlock from Roam

- ````
  # Days since/until Christmas #42SmartBlock
  ````
  - ````
    <%DATEBASISDAILYNOTES%><%NOBLOCKOUTPUT%>
    ````
  - ````
    <%SET:event,Christmas is in %><%NOBLOCKOUTPUT%>
    ````
  - ````
    <%SET:target,<%DATE:December 25, 2020%>%><%NOBLOCKOUTPUT%>
    ````
  - ````
    <%SET:today,<%DATE:today%>%><%NOBLOCKOUTPUT%>
    ````
  - ````
   <%GET:event%><%JAVASCRIPT:```javascript
    var event = roam42.smartBlocks.activeWorkflow.vars["event"];
    var today = roam42.smartBlocks.activeWorkflow.vars["today"];
    var target = roam42.smartBlocks.activeWorkflow.vars["target"];
    console.log(event);
    console.log(today);
    console.log(target);
    today = today.trim().split("[").join("").split("]").join("").replace("#","");
    today = today.replace("st,",",").replace("rd,",",").replace("th,",",").replace("nd,",",");
    var start = new Date(Date.parse(today));
    target = target.trim().split("[").join("").split("]").join("").replace("#","");
    target = target.replace("st,",",").replace("rd,",",").replace("th,",",").replace("nd,",",");
    var end = new Date(Date.parse(target));
    var days = end.getDate() - start.getDate();
    return(days + " days");```%>^^
    ````

## 📋 Describe the SmartBlock

This SmartBlock returns the number of days from today (or the day of the daily note page you add it to) to some event - Chirstmas, Dec 25, in this case.

## 📷 Screenshot of your #42SmartBlock workflow/template from Roam

![image](https://user-images.githubusercontent.com/67427527/101268741-cea9e580-3734-11eb-8b7b-7b6eebdc91ee.png)

## 💡 Additional Info

Update the "event" and "target" date at the top to create a counter for the event you want.

**Comment by roytemple** [comment]
I changed it as suggested: 
- <%SET:event,The Election is in %><%NOBLOCKOUTPUT%>
- <%SET:target,<%DATE:November 8, 2022%>%><%NOBLOCKOUTPUT%>
The text part renders correctly. But it's still counting the number of days until Christmas this year. Any ideas?

**Comment by eatondpe** [comment]
@roytemple Sorry, Roy. Please use this one, instead.

https://github.com/roamhacker/SmartBlocks/issues/113

**Comment by eatondpe** [comment]
@roytemple This has been fixed and the updated code added to here. Thank you for reporting the error.

**Comment by n8dunn** [comment]
@eatondpe Thanks for this. How would I modify this to count a positive number of days since a past event? Example: 16,723 days since I was born, etc.

**Comment by eatondpe** [comment]
> @eatondpe Thanks for this. How would I modify this to count a positive number of days since a past event? Example: 16,723 days since I was born, etc.

@n8dunn - this should do it...

Replace this:
var days = Math.round((end - start) / 1000 / 60 / 60 / 24);

with this:
var days = Math.round((start - end) / 1000 / 60 / 60 / 24);

**Comment by n8dunn** [comment]
> @n8dunn - this should do it...
> 
> Replace this:
> var days = Math.round((end - start) / 1000 / 60 / 60 / 24);
> 
> with this:
> var days = Math.round((start - end) / 1000 / 60 / 60 / 24);

@eatondpe Yep, that worked perfectly! Thanks so much for the simple solution and quick response.
